src/dbnode/storage/namespace.go

```diff
@@ -457,6 +457,14 @@ func (n *dbNamespace) Shards() []Shard {
 }
 
 func (n *dbNamespace) AssignShardSet(shardSet sharding.ShardSet) {
+	n.assignShardSet(shardSet, n.nopts.BootstrapEnabled(), false)
+}
+
+func (n *dbNamespace) assignShardSet(
+	shardSet sharding.ShardSet,
+	needsBootstrap bool,
+	initialAssignment bool,
```

Comment:
Maybe we should create these as options so it's not easy to mix up when passing into the function?
```go
type assignShardSetOptions struct {
  needsBootstrap bool
  initialAssignment bool
}
```